src/stores/explorer.ts

```diff
@@ -98,14 +98,38 @@ export default class Explorer {
   public getAddress(address: string) {
     address = address.toLowerCase();
     const token = this.tokens.tokenForAddress(address);
-    return this.web3.plasma.instance.eth.getBalance(address)
-      .then((balance) => {
-        return {
-          address,
-          token,
-          balance,
-        };
-      });
+    return this.web3.plasma.instance.getUnspent(address).then(unspents => {
+      const colors = Array.from(
+        new Set([0, ...unspents.map(u => u.output.color)])
+      );
+      return Promise.all(
+        colors.map(color => {
+          const token = this.tokens.tokenForColor(color);
+          if (!token) {
+            return Promise.resolve(bi(0));
+          }
+
+          return token.balanceOf(address, true);
+        })
+      )
+        .then(balances => {
+          return colors.reduce<{ [key: string]: any }>(
```

Comment:
🤯

Reply:
Ah, true. I should use real type here, not any :D